woocommerce/woocommerce-ios
# [Mobile Payments] Show Tap to Pay on iPhone feedback row in Payments menu

Closes: #9293


## Description


We want to get feedback from our merchants once they have taken at least one payment with Tap to Pay on iPhone, including the test payment.

This PR adds a button to the Payments menu to allow them to give feedback via a CrowdSignal survey. The button shows for 30 days from their first Tap to Pay on iPhone payment, with a given device and store combination.

Production link: `https://automattic.survey.fm/woo-app-–-first-ttp-survey`
Development link: `https://automattic.survey.fm/woo-app-tap-to-pay-survey`

## Testing instructions


Using a US-based store and an iPhone XS or newer on iOS 16+

1. Delete the app from your phone (not necessary first time, but done to clear any stored dates after that)
2. Navigate to Menu > Payments
3. Observe that the feedback button is not shown
4. Tap `Set up Tap to Pay on iPhone`
5. Follow the flow, and tap `Skip` when prompted to make a test payment
6. Observe that the feedback button is still not shown
7. Tap `Set up Tap to Pay on iPhone` again, and follow the flow
8. This time, complete the test payment
9. Observe that when you dismiss the order at the end of the flow, the `Share Tap to Pay on iPhone Feedback` row is shown
10. Tap the button and complete the survey

Repeat the test using the Simple Payments flow, and the order payments flow.

### Testing that the button disappears after 30 days
Modify [the code to check for thirty days passing](https://github.com/woocommerce/woocommerce-ios/blob/22f0aaf0c3886cf1791539d3c04313ebbab31c3c/WooCommerce/Classes/ViewRelated/Dashboard/Settings/In-Person%20Payments/InPersonPaymentsMenuViewModel.swift#L86), to instead check for 30 seconds, by updating it to the following:
```
    let thirtyDaysAgo = Calendar.current.date(byAdding: DateComponents(second: -30), to: Date()) else {
```

Observe that the feedback button is only shown when you view the payments menu within 30 seconds of taking your first payment.

## Screenshots


https://user-images.githubusercontent.com/2472348/229856673-1088fa39-2de0-48f8-be6c-3681b0ddc55e.mp4



---
